### ISSUE TITLE
Android Messaging - Develop Branch

### ISSUE THREAD

**Issue by johnnymajor** [Wizcorp/cordova-plugin-wizviewmanager#84]
I have not tried pulling from Master because using the latest Cordova so needed to pull from develop for Android support however messaging is not working.

When attempting to send a message to the 'mainView' the application is failing.

**Call**

``` javascript
wizViewMessenger.postMessage({
            requestId : 'test'
        }, 'mainView');
```

**Errors**
01-12 11:19:23.679 21838-21838/com.cnxcorp.portal W/System.err: java.lang.ClassCastException: org.apache.cordova.CordovaWebViewImpl cannot be cast to android.webkit.WebView
01-12 11:19:23.679 21838-21838/com.cnxcorp.portal W/System.err:     at jp.wizcorp.phonegap.plugin.wizViewManager.WizWebView$1.shouldOverrideUrlLoading(WizWebView.java:156)
01-12 11:19:23.679 21838-21838/com.cnxcorp.portal W/System.err:     at com.android.webview.chromium.WebViewContentsClientAdapter.shouldOverrideUrlLoading(WebViewContentsClientAdapter.java:293)
01-12 11:19:23.679 21838-21838/com.cnxcorp.portal W/System.err:     at com.android.org.chromium.android_webview.AwContentsClientBridge.shouldOverrideUrlLoading(AwContentsClientBridge.java:96)
01-12 11:19:23.689 21838-21838/com.cnxcorp.portal W/System.err:     at com.android.org.chromium.base.SystemMessageHandler.nativeDoRunLoopOnce(Native Method)
01-12 11:19:23.689 21838-21838/com.cnxcorp.portal W/System.err:     at com.android.org.chromium.base.SystemMessageHandler.handleMessage(SystemMessageHandler.java:27)
01-12 11:19:23.689 21838-21838/com.cnxcorp.portal W/System.err:     at android.os.Handler.dispatchMessage(Handler.java:102)
01-12 11:19:23.689 21838-21838/com.cnxcorp.portal W/System.err:     at android.os.Looper.loop(Looper.java:146)
01-12 11:19:23.689 21838-21838/com.cnxcorp.portal W/System.err:     at android.app.ActivityThread.main(ActivityThread.java:5598)
01-12 11:19:23.689 21838-21838/com.cnxcorp.portal W/System.err:     at java.lang.reflect.Method.invokeNative(Native Method)
01-12 11:19:23.689 21838-21838/com.cnxcorp.portal W/System.err:     at java.lang.reflect.Method.invoke(Method.java:515)
01-12 11:19:23.689 21838-21838/com.cnxcorp.portal W/System.err:     at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1283)
01-12 11:19:23.689 21838-21838/com.cnxcorp.portal W/System.err:     at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1099)
01-12 11:19:23.689 21838-21838/com.cnxcorp.portal W/System.err:     at dalvik.system.NativeStart.main(Native Method)

Views never receive messages that are sent from the mainView

Messaging is working fine for iOS off the develop branch.

Please let me know if you need any more information. Thanks!!

**Comment by johnnymajor** [comment]
For the children that are attempting to post messages I found that 

wizWebView.java 'shouldOverrideUrlLoading' needed to be changed.
```
WebView targetView;
                        try {
                            targetView = (WebView) viewList.get(msgData[1]);

                            // send data to mainView
                            String data2send = msgData[2];
                            // Log.d("WizWebView", "[wizMessage] targetView ****** is " + msgData[1] + " -> " + targetView + " with data -> " + data2send);
                            if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.KITKAT) {
                                // Only for Kitkat and newer versions
                                targetView.evaluateJavascript("wizViewMessenger.__triggerMessageEvent(\"" + msgData[0] + "\", \"" + msgData[1] + "\", \"" + data2send + "\", \"" + msgData[3] + "\");", null);
                            } else {
                                targetView.loadUrl("javascript:wizViewMessenger.__triggerMessageEvent(\"" + msgData[0] + "\", \"" + msgData[1] + "\", \"" + data2send + "\", \"" + msgData[3] + "\");");
                            }
                        } catch (JSONException e) {
                            // TODO Auto-generated catch block
                            e.printStackTrace();
                        }
```

The above was failing on : 
```
targetView = (WebView) viewList.get(msgData[1]);
```

Changed it to use the Cordova webview and run the javascript with 'sendJavascript'. It works however 'sendJavascript' is deprecated.
```
try {
                            CordovaWebView targetView = (CordovaWebView) viewList.get(msgData[1]);

                            // send data to mainView
                            String data2send = msgData[2];
                            // Log.d("WizWebView", "[wizMessage] targetView ****** is " + msgData[1] + " -> " + targetView + " with data -> " + data2send);
                            targetView.sendJavascript("wizViewMessenger.__triggerMessageEvent(\"" + msgData[0] + "\", \"" + msgData[1] + "\", \"" + data2send + "\", \"" + msgData[3] + "\");");
                        } catch (JSONException e) {
                            // TODO Auto-generated catch block
                            e.printStackTrace();
                        }
```

**Comment by johnnymajor** [comment]
Anyone have any idea's or guidance on this one? Thanks

**Comment by jrouault** [comment]
Hello, sorry we didn't have time to look at your issue before. There is some indications on how to fix this in Cordova-Android's doc:
https://github.com/apache/cordova-android/blob/master/framework/src/org/apache/cordova/CordovaWebView.java#L70
These changes might affect the basics of this plugin but if you feel confident, please try it out.

**Comment by jrouault** [comment]
Closing as this should have been fixed with #90. Please re-open the issue if you still have it after updating.
